relay-general/src/store/normalize.rs

```diff
@@ -82,6 +84,59 @@ pub fn normalize_dist(dist: &mut Option<String>) {
     }
 }
 
+/// Compute additional measurements derived from existing ones
+pub fn compute_measurements(transaction_duration_ms: f64, measurements: &mut Measurements) {
+    if let Some(frames_total) = measurements.get_value("frames_total") {
+        if frames_total > 0.0 {
+            if let Some(frames_frozen) = measurements.get_value("frames_frozen") {
+                measurements.insert(
+                    "frames_frozen_rate".to_owned(),
+                    Measurement {
+                        value: Annotated::new(frames_frozen / frames_total),
+                        unit: Annotated::new(MetricUnit::Fraction(FractionUnit::Ratio)),
+                    }
+                    .into(),
+                );
+            }
+            if let Some(frames_slow) = measurements.get_value("frames_slow") {
+                measurements.insert(
+                    "frames_slow_rate".to_owned(),
+                    Measurement {
+                        value: Annotated::new(frames_slow / frames_total),
+                        unit: Annotated::new(MetricUnit::Fraction(FractionUnit::Ratio)),
+                    }
+                    .into(),
+                );
+            }
+        }
+    }
+
+    // Get stall_percentage
+    if transaction_duration_ms > 0.0 {
+        if let Some(stall_total_time) = measurements
+            .get("stall_total_time")
+            .and_then(Annotated::value)
+        {
+            if stall_total_time.unit.value()
+                == Some(&MetricUnit::Duration(DurationUnit::MilliSecond))
+            {
+                if let Some(stall_total_time) = stall_total_time.value.0 {
+                    measurements.insert(
+                        "stall_percentage".to_owned(),
+                        Measurement {
+                            value: Annotated::new(
+                                100.0 * (stall_total_time / transaction_duration_ms),
+                            ),
+                            unit: Annotated::new(MetricUnit::Fraction(FractionUnit::Percent)),
```

Comment:
@wmak I multiply by 100 and set the unit `Percent` here, because that is what the name `percentage` seems to imply. But looking at https://github.com/getsentry/sentry/blob/3b3cb01fd16e9ccce104a077efa29a89d8253891/src/sentry/search/events/datasets/discover.py#L1222-L1225, the product actually expects a ratio between 0 and 1?

```python
    def _resolve_measurements_stall_percentage(self, _: str) -> SelectType:
        return self._resolve_aliased_division(
            "measurements.stall_total_time", "transaction.duration", MEASUREMENTS_STALL_PERCENTAGE
        )

```

Reply:
The definition in search fields makes this a ratio like all others without a multiplication by 100. The product type of "percentage" is actually 0-1. See: https://github.com/getsentry/sentry/blob/8b13ba0e01744d88632d6b8cff185f5644b40c0a/src/sentry/search/events/fields.py#L414-L426

Reply:
ACK, will change the unit.

Reply:
+1 Jan is correct here, but there's no strong reason for 0-1 or 0-100 it was just an arbitrary decision. So if there was a good reason to we could likely update to 0-100.